internal/cmd/gocdk/launch.go

```diff
@@ -15,11 +15,170 @@
+	}
+
+	// Get the image name from the Dockerfile if not specified.
+	if *imageName == "" {
+		var err error
+		*imageName, err = moduleDockerImageName(moduleRoot)
+		if err != nil {
+			return xerrors.Errorf("gocdk launch: %w", err)
+		}
+	}
+
+	// Prepare the launcher.
+	cfg, err := readBiomeConfig(moduleRoot, biome)
+	if err != nil {
+		return xerrors.Errorf("gocdk launch: %w", err)
+	}
+	if cfg.Launcher == nil {
+		return xerrors.Errorf("gocdk launch: launcher not specified in %s", filepath.Join(findBiomeDir(moduleRoot, biome), biomeConfigFileName))
+	}
+	launcher, err := newLauncher(ctx, pctx, *cfg.Launcher)
+	if err != nil {
+		return xerrors.Errorf("gocdk launch: %w", err)
+	}
+
+	// Read the launch specifier from the biome's Terraform output.
+	tfOutput, err := tfReadOutput(ctx,
+		findBiomeDir(moduleRoot, biome),
+		pctx.overrideEnv("TF_IN_AUTOMATION=1"))
+	if err != nil {
+		return xerrors.Errorf("gocdk launch: %w", err)
+	}
+
+	// Launch the application.
+	launchURL, err := launcher.Launch(ctx, &launchInput{
+		dockerImage: *imageName,
+		specifier:   launchSpecifier(tfOutput["launch_specifier"].mapValue()),
+		// TODO(light): Pass environment variables from tfOutput into env.
+	})
+	if err != nil {
+		return xerrors.Errorf("gocdk launch: %w", err)
+	}
+	fmt.Fprintf(pctx.stdout, "Serving at %s\n", launchURL)
+	return nil
+}
+
+// Launcher is the interface for any type that can launch a Docker image.
+type Launcher interface {
+	Launch(ctx context.Context, input *launchInput) (*url.URL, error)
+}
+
+// launchInput is the input to a launcher.
+type launchInput struct {
+	// dockerImage specifies the image name and tag of the local Docker image to
+	// deploy. If the local image does not exist, then the launcher should return
+	// an error.
+	dockerImage string
+
+	// env is the set of additional environment variables to set. It should not
+	// include PORT nor should it contain multiple entries for the same variable
+	// name.
+	env []string
+
+	// specifier is the set of arguments passed from a biome's Terraform module.
+	specifier launchSpecifier
+}
+
+// newLauncher creates the launcher for the given name.
+func newLauncher(ctx context.Context, pctx *processContext, launcherName string) (Launcher, error) {
+	logger := log.New(pctx.stderr, "gocdk: ", log.Ldate|log.Ltime)
+	switch launcherName {
+	case "local":
+		return &localLauncher{
+			logger:    logger,
+			dockerEnv: pctx.env,
+			dockerDir: pctx.workdir,
+		}, nil
+	default:
+		return nil, xerrors.Errorf("prepare launcher: unknown launcher %q", launcherName)
+	}
+}
+
+// localLauncher starts local Docker containers.
+type localLauncher struct {
+	logger    *log.Logger
+	dockerEnv []string
+	dockerDir string
+}
+
+// Launch implements Launcher.Launch.
+func (local *localLauncher) Launch(ctx context.Context, input *launchInput) (*url.URL, error) {
+	dockerArgs := []string{
+		"run",
+		"--rm",
+		"--detach",
+		"--publish", "8080:8080",
```

Comment:
it might be nice for it to be settable, but I think it's ok for running `serve` and `deploy` in the same biome simultaneously to be mutually exclusive.

Reply:
Added a launch specifier option.